src/main/java/seedu/address/model/event/StartTime.java

```diff
@@ -3,55 +3,55 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import java.text.DecimalFormat;
+import java.time.DateTimeException;
 import java.time.LocalTime;
-import java.time.format.DateTimeFormatter;
-import java.time.format.DateTimeParseException;
+
 
 /**
  * Represents an Event's starting time in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidStartTime(String)}
  */
 public class StartTime implements Comparable<StartTime> {
 
-    public static final String MESSAGE_CONSTRAINTS = "Start time must be in format: hh/mm/AM, hh/mm/PM or hh:mm";
-
-    //for checking if valid input date format
-    private static final DateTimeFormatter checkFormatter = DateTimeFormatter
-            .ofPattern("[HH:MM][hh/mm/a][h/mm/a][hh/m/a][h/m/a]");
-
-    //for changing to storage friendly format
-    private static final DateTimeFormatter logFormatter = DateTimeFormatter.ofPattern("hh/mm/a");
-
-    //for changing to user-readable format
-    private static final DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("hh:mm a");
-
+    public static final String MESSAGE_FORMAT_CONSTRAINTS = "Start time must be in format: hh:mm.";
+    public static final String MESSAGE_VALUE_CONSTRAINTS = "%s exceeds the range of valid time value.";
+    private static final DecimalFormat df = new DecimalFormat("00");
     public final LocalTime time;
 
 
     /**
      * Constructs a {@code StartTime}.
-     *
      * @param startTime A valid start time.
      */
     public StartTime(String startTime) {
         requireNonNull(startTime);
         assert !startTime.isBlank();
-        checkArgument(isValidStartTime(startTime), MESSAGE_CONSTRAINTS);
-        this.time = LocalTime.parse(startTime, checkFormatter);
+        checkArgument(isValidStartTime(startTime) != 0, MESSAGE_FORMAT_CONSTRAINTS);
+        checkArgument(isValidStartTime(startTime) != -1, String.format(MESSAGE_VALUE_CONSTRAINTS, startTime));
+        String [] parsedTime = startTime.split(":", 2);
+        this.time = LocalTime.of(Integer.parseInt(parsedTime[0]), Integer.parseInt(parsedTime[1]), 0);
```

Comment:
Java actually provides a `LocalTime.of()` method that sets the "seconds" param to 0 by default! 😄

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/LocalTime.html#of(int,int)

```suggestion
        this.time = LocalTime.of(Integer.parseInt(parsedTime[0]), Integer.parseInt(parsedTime[1]));
```